exercicios/para-casa/index.js

```diff
@@ -0,0 +1,69 @@
+const express = require('express')
+const app = express()
+const listaDeUsuarios = require("./model/usuarios.json")
+const port = 3004//rota diferente para evitar conflitos
+
+app.use(express.json())
+
+
+// [x] Rota que atualiza todos os dados de cadastro 
+// de um usuário e se não for encontrado cria um novo na lista
+app.put("/usuarios/:id", (req, res) => {
+const idUsuario = req.params.id
+const idAtualizado = req.body
+
+const usuario = listaDeUsuarios.find(usuario => usuario.id == idUsuario)
+
+if(usuario) {
+  const usuarioAtualizado = {
+    ...usuario,
+    ...idAtualizado
+}
+for(let usuario of listaDeUsuarios){ // percorre objetos, incluindo Array 
+  if(usuario.id == idUsuario){
+    listaDeUsuarios[usuario] = usuarioAtualizado
```

Comment:
Isso aqui não vai funcionar como você espera.

Para acessar um item em um array você precisa passar um número , assim você acessa esse objeto no Array de objetos. Dessa forma que você fez foi declarado um objeto como index, logo, o registro ficará assim na lista:

```javascript
'[object Object]': {
    id: 18,
    name: 'Jacob Ingram',
    address: {
      zipcode: '12333-000',
      street: 'Rua São Sebastião',
      city: 'Capixabaa'
    },
    profession: 'Stress Reduction Specialist',
    birth_date: '1965-06-01T23:54:30.208Z'
  }
```

TIP: o `.map()` é muito útil pois ele consegue percorrer todo o array de objetos,a cessar o objeto e também o index daquele objeto no array. 

Usando sua solução: um simples `for` resolveria.